src/app/script/sprite_class.cpp

```diff
@@ -710,12 +711,19 @@ int Sprite_deleteTile(lua_State* L)
   auto sprite = get_docobj<Sprite>(L, 1);
   tile_index ti = 0;
   auto ts = may_get_docobj<Tileset>(L, 2);
+  std::string temp = " ";
   if (ts)
     ti = lua_tointeger(L, 3);
-  else
+  else {
     ts = get_tile_index_from_arg(L, 2, ti);
+    temp = " on the TileObj";
+  }
   if (!ts)
     return luaL_error(L, "Sprite:deleteTile() needs a Tileset or Tile as first argument");
+  if (ts->sprite() != sprite)
+    return luaL_error(L, "the tileset%s belongs to another sprite", temp.c_str());
```

Comment:
Probably we can simplify the error message just to avoid creating a std::string, e.g.
```suggestion
    return luaL_error(L, "the tileset belongs to another sprite");
```

Reply:
No problem. Done!